avocado/utils/ssh.py

```diff
@@ -248,3 +249,22 @@ def copy_files(self, source, destination, recursive=False):
             return result.exit_status == 0
         except process.CmdError as exc:
             raise NWException("failed to copy file {}".format(exc))
+
+    def remove_control_path_reconnect(self):
+        """
+        Establishes connection to the remote host if failed try to
+        close the session.if it failed remove control path file and
+        reconnect it.
```

Comment:
Now that I'm reading this docstring, this function sounds really strange to me. Let me explain:

Users of this function will try to connect to a host and if not possible you are removing ***all*** control path, affecting other connections, for instance. 

If you would like to remove the ***specific*** control path for that connection that failed, IMO makes sense to have this inside the "connect()" function.